lib/elixir/src/elixir_with.erl

```diff
@@ -57,3 +78,21 @@ build_cases([], DoExpr) ->
   DoExpr;
 build_cases([{_, Meta, _} = Expr | Rest], DoExpr) ->
   {'__block__', Meta, [Expr, build_cases(Rest, DoExpr)]}.
+
+build_else(Meta, WithCases, ElseClauses) ->
+  Result = {'result', Meta, nil},
+  Clauses = [
+    {'->', Meta, [[{{'^', Meta, [?ref(Meta)]}, Result}], Result]}
+    | ElseClauses
+  ] ++ [build_raise(Meta)],
+  {'case', Meta, [WithCases, [{do, Clauses}]]}.
+
+reference(Meta) ->
```

Comment:
Also, we need to write tests about variables inside the `else` part, as they also should never leak. Check "lib/elixir/test/elixir/kernel/expansion_test.exs" for tests that check the variable scope.